### PR TITLE
feat: added way to skip zone creation

### PR DESCRIPTION
**Pull request by Ak-sky**
### Description

Added ability to skip specific service zone creation.

### Types of changes in this PR
https://github.com/terraform-ibm-modules/terraform-ibm-cbr/issues/269
#### Changes that affect the core Terraform module or submodules
- [ ] Bug fix
- [x] New feature
- [ ] Dependency update

#### Changes that don't affect the core Terraform module or submodules
- [x] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI-related update (pipeline, etc.)
- [ ] Other

#### Release required?
Identify the type of release. For information about the changes in a semantic versioning release, see [Release versioning](https://terraform-ibm-modules.github.io/documentation/#/versioning).

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

If a release is required, replace this text with information that users need to know about the release. Write the release notes to help users understand the changes, and include information about how to update from the previous version.

Your notes help the merger write the commit message for the PR that is published in the release notes for the module.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### Merge actions for mergers

- Use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents and any release notes provided by the PR author. The commit message determines whether a new version of the module is needed, and if so, which semver increment to use (major, minor, or patch).
- Merge by using "Squash and merge".

### CODE REVIEW

File: examples/fscloud/variables.tf [hunk 1]
```diff
@@ -57,3 +57,9 @@ variable "allow_vpcs_to_cos" {
   description = "Set rule for VPCs to COS, deafult is true"
   default     = true
 }
+
+variable "skip_specific_services_for_zone_creation" {
```
Comment (vburckhardt) [inline examples/fscloud/variables.tf:61]: I'd suggest to add the default values inline in the main as opposed to a variable, along with some comment indicating that this is for illustration purposes ( we're moving away from having input variables in the examples )

File: examples/fscloud/main.tf [hunk 2]
```diff
@@ -105,7 +108,7 @@ module "cbr_zone_operator_ips" {
 }
 
 ## Examples of data lookup on objects (zone, rule) created by the fscoud profile module
-## Get rule targetting "event-notification"
+# ## Get rule targetting "event-notification"
```
Comment (vburckhardt) [inline examples/fscloud/main.tf:111]: Extra # is not needed

**Review by vburckhardt — COMMENTED** [review]
Thanks for this PR. I put some small comments, and then I think this can go. Could you also just describe how this was tested.

File: examples/fscloud/variables.tf (hunk 1 above)
Comment (Ak-sky) [reply examples/fscloud/variables.tf:61]: done

File: examples/fscloud/main.tf (hunk 2 above)
Comment (Ak-sky) [reply examples/fscloud/main.tf:111]: removed

**Comment by Ak-sky** [comment]
> Thanks for this PR. I put some small comments, and then I think this can go. Could you also just describe how this was tested.

This was tested by running the fscloud example where -
   1- 2 service references `skip_specific_services_for_zone_creation = ["user-management", "iam-groups"]` were used to skip creation of zones, verified in the output. 
[Skip_ref_TF_apply_08.08.2023-01.16.33.log](https://github.com/terraform-ibm-modules/terraform-ibm-cbr/files/12284845/Skip_ref_TF_apply_08.08.2023-01.16.33.log)

   2- Empty list for `skip_specific_services_for_zone_creation = []` to verify if the zone were created for these 2 service references ("user-management", "iam-groups").
[Empty_list_sr_TF_apply_08.08.2023-01.28.21.log](https://github.com/terraform-ibm-modules/terraform-ibm-cbr/files/12284846/Empty_list_sr_TF_apply_08.08.2023-01.28.21.log)

File: examples/fscloud/main.tf [hunk 3]
```diff
@@ -55,6 +55,9 @@ module "cbr_account_level" {
   allow_vpcs_to_container_registry = var.allow_vpcs_to_container_registry
   allow_vpcs_to_cos                = var.allow_vpcs_to_cos
 
+  # Demonstrates how zone creation will be skipped for these two service references ["user-management", "iam-groups"]
```
Comment (vburckhardt) [inline examples/fscloud/main.tf:58]: I think we should also put a note in the README of the example to indicate this.

Comment (Ak-sky) [reply examples/fscloud/main.tf:58]: done

**Review by vburckhardt — DISMISSED** [review]

**Review by vburckhardt — APPROVED** [review]

**Comment by ocofaigh** [comment]
@Ak-sky The release notes in https://github.com/terraform-ibm-modules/terraform-ibm-cbr/releases/tag/v1.7.0 are missing some important changes that were made in this PR that consumers need to know. Such as the movement of the submodules in a modules folder. Consumers now need to update their code to consume latest version, so all these details are needed in release notes. For example, in previous versions consumers used this:
`source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//cbr-zone-module?ref=v1.6.1"`

Now then need to use:
`"git::https://github.com/terraform-ibm-modules/terraform-ibm-cbr//modules/cbr-zone-module?ref=v1.7.0"` 

@Ak-sky can you summarise ALL the changes made in this PR so we can update release notes please?

**Comment by Ak-sky** [comment]
Hi @ocofaigh, I have update the release notes, please have a look once.

**Pull request merged**